pkgs/development/compilers/ghc/9.2.1.nix

```diff
@@ -144,6 +144,14 @@ stdenv.mkDerivation (rec {
     sha256 = "1q2pppxv2avhykyxvyq72r5p97rkkiqp19b77yhp85ralbcp4ivw";
   };
 
+  patches = [
+    (fetchpatch {
+      name = "fix-darwin-link-failure.patch";
+      url = "https://gitlab.haskell.org/ghc/ghc/-/commit/77456387025ca74299ecc70621cbdb62b1b6ffc9.patch";
+      sha256 = "1g8smrn7hj8cbp9fhrylvmrb15s0xd8lhdgxqnx0asnd4az82gj8";
+    })
```

Comment:
@sternenseemann Could you write a comment about why this is needed?

I guess it is https://gitlab.haskell.org/ghc/ghc/-/issues/19950#note_373726?